siflower: add testing kernel 6.12

libiunc opened the pull request:
I upgraded the kernel from 6.6 to 6.12 for testing.

Compile and test:

1. Bananapi BPi-RV2 (Booting from NAND)
2. Siflower SF19A2890 EVB

Run tested:

1. Bananapi BPi-RV2 (Booting from NAND)

libiunc committed:
kernel/siflower: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc committed:
kernel/siflower: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

981213 commented on target/linux/siflower/files-6.12/drivers/pci/controller/dwc/pcie-sf21.c:
```diff
@@ -0,0 +1,358 @@
+}
+
+static const struct dw_pcie_host_ops sf_pcie_host_ops = {
+	.post_init = sf_pcie_host_init,
```
This is apparently wrong. When doing kernel-bump you should check what's actually changed between kernel versions, usually by tracking down the exact commit introducing the API changes. The commit introducing API change will come with the fix for in-tree drivers, which you can follow to fix any out-of-tree drivers.

Blindly following compiler suggestions doesn't really work.

libiunc replied:
done

LGA1150 commented on target/linux/siflower/sf21/config-6.12:
```diff
@@ -49,6 +53,7 @@ CONFIG_CRYPTO_LIB_UTILS=y
 CONFIG_CRYPTO_ZSTD=y
 CONFIG_DEBUG_BUGVERBOSE=y
 CONFIG_DEBUG_INFO=y
+# CONFIG_KERNEL_UNCOMPRESSED is not set
```
Were you manually editing this file? Please use `make kernel_menuconfig` instead.

LGA1150 commented on target/linux/siflower/sf21/config-6.12:
```diff
@@ -217,16 +223,22 @@ CONFIG_RISCV=y
 CONFIG_RISCV_ISA_SVNAPOT=y
 CONFIG_RISCV_ISA_SVPBMT=y
 # CONFIG_RISCV_ISA_V is not set
+CONFIG_RISCV_ISA_ZAWRS=y
```
C908 does not support Zawrs, so this can be disabled.

LGA1150 commented on target/linux/siflower/sf21/config-6.12:
```diff
@@ -217,16 +223,22 @@ CONFIG_RISCV=y
+CONFIG_RISCV_ISA_ZAWRS=y
+CONFIG_RISCV_ISA_ZBA=n
 CONFIG_RISCV_ISA_ZBB=y
+CONFIG_RISCV_ISA_ZBC=n
```
C908 supports Zba and Zbc, so these should be enabled.

LGA1150 commented on target/linux/siflower/sf21/config-6.12:
```diff
@@ -284,3 +296,8 @@ CONFIG_ZONE_DMA32=y
+CONFIG_RISCV_ISA_VENDOR_EXT_ANDES=y
+CONFIG_BUILTIN_DTB=n
+CONFIG_SERIAL_EARLYCON_RISCV_SBI=n
+CONFIG_HVC_RISCV_SBI=y
```
This is for guest OS in VM. Do not enable.

Ansuel commented:
@libiunc i know it's a lot to ask but hope you can improve this

Please split this to refresh for 6.6 and 6.12. Also split commit that drop upstream patch from commit that make manual rebase of the patch.

And please also split the driver changes that fix trivial stuff (like static or other minor changes) than changes that adapt to new api for 6.12

udanfacy22 commented:
I started a forum thread requesting 6.12 run testing for this and other targets: https://forum.openwrt.org/t/kernel-6-12-debugging-and-testing-needed-for-at91-bcm47xx-bcm4908-bcm53xx-mpc85xx-qoriq-siflower-and-zynq/242857

981213 commented on target/linux/siflower/files-6.12/drivers/net/ethernet/siflower/sfxgmac-ext.h:
```diff
@@ -28,4 +29,9 @@ struct gmac_common {
+struct xgmac_dma_priv;
+
+extern netdev_tx_t xgmac_dma_xmit(struct sk_buff *skb, struct xgmac_dma_priv *priv);
+extern void xgmac_dma_disable(struct xgmac_dma_priv *priv);
```
why is this needed?

981213 commented on target/linux/siflower/files-6.12/drivers/net/ethernet/siflower/sfxgmac.c:
```diff
@@ -980,39 +957,39 @@ static int xgmac_ethtool_get_link_ksettings(struct net_device *dev,
-	ret = phylink_ethtool_get_eee(priv->phylink, e);
-	if (ret)
-		return ret;
+    ret = phylink_ethtool_get_eee(priv->phylink, eee);
```
indent with tab instead.

981213 reviewed (changes requested):
Hello! more comments below:

libiunc commented on target/linux/siflower/files-6.12/drivers/net/ethernet/siflower/sfxgmac.c:
```diff
@@ -980,39 +957,39 @@ static int xgmac_ethtool_get_link_ksettings(struct net_device *dev,
 }
 
-static int xgmac_ethtool_get_eee(struct net_device *dev, struct ethtool_eee *e)
+static int xgmac_ethtool_get_eee(struct net_device *dev, struct ethtool_keee *eee)
```
I referenced [this](https://github.com/openwrt/openwrt/blob/main/target/linux/generic/backport-6.12/721-01-v6.15-net-ethernet-mediatek-add-EEE-support.patch#L46)

libiunc committed:
siflower: 6.12: drop upstreamed patches

Drop patches from the 6.12 patchset that have been merged into the
mainline Linux kernel since the previous version.

Dropped patches:
001-net-phy-c45-add-genphy_c45_pma_read_ext_abilities-fu.patch
002-net-phy-Optimize-phy-speed-mask-to-be-compatible-to-.patch
003-net-phy-Add-driver-for-Motorcomm-yt8821-2.5G-etherne.patch
014-riscv-add-an-option-for-efficient-unaligned-access.patch

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc committed:
siflower: 6.12: refresh patches

This involves rebasing the patches and resolving any conflicts
arising from changes in the base kernel between the previous
supported version and 6.12.

Tested-by: Jingkun Zheng <mx@kevinmx.top>
Tested-by: Zhu Yujie <libriunc@gmail.com>
Tested-by: Chuanhong Guo gch981213@gmail.com
Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc committed:
siflower: spi: replace 'master' with 'controller' in driver

This change replaces the term 'master' with 'controller' in the
Siflower SPI driver (sf_qspi) to align with the Linux kernel's
inclusive language policy.

Tested-by: Jingkun Zheng <mx@kevinmx.top>
Tested-by: Zhu Yujie <libriunc@gmail.com>
Tested-by: Chuanhong Guo gch981213@gmail.com
Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc commented:
> @libiunc i know it's a lot to ask but hope you can improve this
> 
> Please split this to refresh for 6.6 and 6.12. Also split commit that drop upstream patch from commit that make manual rebase of the patch.
> 
> And please also split the driver changes that fix trivial stuff (like static or other minor changes) than changes that adapt to new api for 6.12

done

Ansuel commented on target/linux/siflower/files-6.12/drivers/gpio/gpio-siflower.c:
```diff
@@ -317,12 +317,11 @@ static int sf_gpio_probe(struct platform_device *pdev)
 }
 
-static int sf_gpio_remove(struct platform_device *pdev)
+static void sf_gpio_remove(struct platform_device *pdev)
```
move this to remove_new first in a separate commit so we can reduce patch delta and also align 6.6 to this change. hope it's not a problem for you!

libiunc committed:
siflower: add support for Linux 6.12

Set KERNEL_TESTING_PATCHVER to allow testing Linux 6.12.

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc committed:
siflower: sf21: update config-6.12

Refresh kernel configuration of the sf21 subtarget for use with Linux 6.12.

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

libiunc committed:
siflower: 6.12: apply trivial driver fixes

Apply trivial fixes for 6.12 files:
- Make functions static
- Add lots of missing header in drivers
- Fix no newline at end of file error and other new line error
- Drop deprecated validate function for xgmac
- Use devm for stmmac probe
- Remove host_ prefix from pcie_host_ops

Tested-by: Jingkun Zheng <mx@kevinmx.top>
Tested-by: Zhu Yujie <libriunc@gmail.com>
Tested-by: Chuanhong Guo gch981213@gmail.com
Signed-off-by: Zhu Yujie <libriunc@gmail.com>
[ improve commit description ]
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
@libiunc please refresh this and move every entry of .remove_new to .remove for kernel 6.12

Ansuel commented:
> This change (.remove_new >> .remove) is not needed,It exists in the target using the 6.12 kernel. Compilation works with gcc15, but gcc14 (using default configuration) causes linux-atm errors due to a FORTIFY issue.

also had a problem with fortify lately... I suggest to clean both host hostpkg and toolchain since you might have dirty fortify headers (they were reverted to a previous version)

But it should be unrelated.

libiunc committed:
siflower: 6.12: change .remove_new callback to .remove

Move back to .remove from .remove_new as kernel 6.12 unified the usage
of .remove and .remove_new (and .remove_new is expected to be dropped in
later version)

Signed-off-by: Zhu Yujie <libriunc@gmail.com>
[ improve commit description ]
Link: https://github.com/openwrt/openwrt/pull/20555
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel commented:
Nope I will merge when I come back home. Thanks a lot fir all the work!

Il Mer 19 Nov 2025, 17:44 libriunk ***@***.***> ha scritto:

> *libiunc* left a comment (openwrt/openwrt#20555)
> <https://github.com/openwrt/openwrt/pull/20555#issuecomment-3553698090>
>
> @Ansuel <https://github.com/Ansuel> Is there anything else I need to
> test? I'm going to sleep now
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/20555#issuecomment-3553698090>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQQXHZCHVWGJALQ6PGT35SM7VAVCNFSM6AAAAACKH57B4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKNJTGY4TQMBZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
